js/ui/html_editor/modules/tableResizing.js

```diff
@@ -0,0 +1,491 @@
+            each($tables, (index, table) => {
+                const $table = $(table);
+                const actualTableWidth = $table.outerWidth();
+                const lastTableWidth = this._getWidthAttrValue($table);
+                if(Math.abs(actualTableWidth - lastTableWidth) > 1) {
+                    $table.attr('width', actualTableWidth + 'px');
+                    this._recalculateColumnsWidth($table);
+                }
+            });
+            this._updateFramesPositions();
+            this._updateFramesSeparators();
+        });
+    }
+
+    _findTables() {
+        return $(this._quillContainer).find('table');
+    }
+
+    _getWidthAttrValue($element) {
+        const attrValue = $element.attr('width');
+        return attrValue ? parseInt(attrValue.replace('px', '')) : undefined;
+    }
+
+    _fixTablesWidths($tables) {
+        each($tables, (_, table) => {
+            const $table = $(table);
+            const $columnElements = this._getTableDeterminantElements($table, 'horizontal');
+            if($columnElements.eq(0).attr('width')) {
+                let columnsSum = 0;
+
+                each($columnElements, (_, element) => {
+                    const $element = $(element);
+                    const columnWidth = $element.attr('width') ? this._getWidthAttrValue($element) : $element.outerWidth();
+                    columnsSum += columnWidth;
+                });
+
+                $table.css('width', 'initial');
+
+                const tableWidth = $table.attr('width') ? parseInt($table.attr('width')) : $table.outerWidth();
+
+                $table.attr('width', Math.max(columnsSum, tableWidth) + 'px');
+            }
+        });
+    }
+
+    _createResizeFrames($tables) {
+        $tables.each((index, $item) => {
+            const $table = $($item);
+            this._tableResizeFrames[index] = {
+                $frame: this._createTableResizeFrame($item),
+                $table: $table,
+                index: index,
+                columnsCount: this._getTableDeterminantElements($table, 'horizontal').length,
+                rowsCount: this._getTableDeterminantElements($table, 'vertical').length
+            };
+        });
+    }
+
+    _isTableChanges() {
+        const $tables = this._findTables();
+
+        let result = false;
+        if($tables.length !== this._tableResizeFrames.length) {
+            result = true;
+        } else {
+            each($tables, (_, table) => {
+                const $table = $(table);
+                const frame = this._getFrameForTable($table);
+
+                if(!frame || frame.columnsCount !== this._getTableDeterminantElements($table, 'horizontal').length || frame.rowsCount !== this._getTableDeterminantElements($table, 'vertical').length) {
+                    result = true;
+                    return false;
+                }
+            });
+        }
+        return result;
+    }
+
+    _getFrameForTable($table) {
```

Comment:
I suggest add some hash table to get appropriate frame for table with `O(1)`, not `O(frames count)`